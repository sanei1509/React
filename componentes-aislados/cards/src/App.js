function App() {
  return (
    <>
      <h1>Creando posteo y tarjetas en react</h1>
    </>
  );
}

export default App;
