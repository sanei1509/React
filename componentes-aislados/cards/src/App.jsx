import NavBar from "./components/nav/NavBar";
// Componentes que vas a usar
import "../src/components/nav/NavBar.css";
import image from "./assets/logo.png";
import "./index.css";

function App() {
  return (
    <>
      <div className="Logo-container p-10">
        <a href="#Home" className="link-logo">
          <img src={image} alt="logo" width={160} />
        </a>
      </div>
      <NavBar />
    </>
  );
}

export default App;
