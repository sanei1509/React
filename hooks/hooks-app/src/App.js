import { Contador } from "./components/Contador.jsx";

function App() {
  return (
    <div className="App">
      <h1>React Hooks</h1>
      <Contador />
    </div>
  );
}

export default App;
