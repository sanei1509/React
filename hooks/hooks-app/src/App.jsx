// import ContadorCustomHook from "./components/ContadorCustomHook";
import "./useReducer/introReducer";
import TodoApp from "./useReducer/TodoApp";
import SimpleForm from "./components/SimpleForm";
import BreakingBad from "./components/BreakingBad";

function App() {
  return (
    <div className="App">
      <h1>React Hooks</h1>
      {/* <Contador /> */}
      {/* <ContadorCustomHook /> */}
      {/* introReducer(); */}
      {/* <SimpleForm /> */}
      {/* <TodoApp /> */}
      <BreakingBad />
    </div>
  );
}

export default App;
