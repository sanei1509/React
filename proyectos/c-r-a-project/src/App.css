@tailwind base;
@tailwind components;
@tailwind utilities;

* {
  box-sizing: border-box;
}

body {
  background-image: linear-gradient(to right, #434343 0%, black 100%);
  background-image: linear-gradient(to right, #434343 0%, black 100%);
  align-items: center;
  padding: 20px;
}

form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: auto;
  justify-content: center;
  align-items: center;
}

.heading-form {
  @apply text-center text-3xl font-bold text-zinc-100;
}

label {
  @apply text-start text-base font-semibold text-zinc-100;
}

.input-container {
  @apply flex flex-col justify-center items-center mt-5;
}

.input-register {
  @apply border-2 border-b-slate-700 p-4 h-8 rounded-lg outline-none shadow-2xl;
}

.button-register {
  @apply flex items-center justify-center mt-4 border-b-slate-700 p-4 w-1/5 h-12 rounded-lg bg-zinc-700 text-xl font-bold text-white;
}
