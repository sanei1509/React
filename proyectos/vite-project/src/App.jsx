import { useState } from "react";
import "./index.css";

// importo los componentes que voy a utilizar
import Header from "./componentes/Header";
import Formulario from "./componentes/Formulario";
import ListadoPacientes from "./componentes/ListadoPacientes";

function App() {
  // podemos trabajar l√≥gica por aqui

  return (
    <div className="App" style={{ width: "100%", alignContent: "center" }}>
      <Header />
      <Formulario />
      <ListadoPacientes />
    </div>
  );
}

export default App;
