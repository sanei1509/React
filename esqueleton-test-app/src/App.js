import "./App.css";
import ContentLoader from "react-content-loader";

function App(props) {
  return (
    <div className="App">
      <h1>Esqueleton Test App</h1>
      <ContentLoader
        speed={2}
        width={400}
        height={460}
        viewBox="0 0 400 460"
        backgroundColor="#404040"
        foregroundColor="#ecebeb"
        {...props}
      >
        <circle cx="37" cy="151" r="23" />
        <rect x="80" y="140" rx="2" ry="2" width="140" height="12" />
        <rect x="79" y="164" rx="2" ry="2" width="140" height="11" />
        <rect x="10" y="193" rx="2" ry="2" width="470" height="177" />
        <rect x="12" y="392" rx="0" ry="0" width="93" height="17" />
        <rect x="131" y="408" rx="0" ry="0" width="29" height="0" />
        <rect x="292" y="389" rx="0" ry="0" width="108" height="17" />
      </ContentLoader>
    </div>
  );
}

export default App;
