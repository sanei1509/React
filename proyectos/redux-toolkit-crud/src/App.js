import "./App.css";
// USEDISPATCH -> permite ejecutar acciones
// USETHUNK -> permite ejecutar acciones asincronas
// USESELECTOR -> permite obtener datos del store
import { TasksList } from "./components/TasksList";
import { TaskForm } from "./components/TaskForm";
import { Route, Routes } from "react-router-dom";

function App() {
  return (
    <div className="App">
      <h1>Gestor de tareas con redux (tkt)</h1>
      <Routes>
        <Route path="/" element={<TasksList />} />
        <Route path="/create" element={<TaskForm />} />
      </Routes>
    </div>
  );
}

export default App;
