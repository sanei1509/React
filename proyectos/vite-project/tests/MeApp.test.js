import { render } from "@testing-library/react";
import App from "../src/MeApp.jsx";

// PRUEBAS MÁS QUE NADA SOBRE EL DOM
describe("pruebas sobre componente principal REACT MeApp.js", () => {
  test("test DOM", () => {
    const title = "Welcome Natalia Oreiro!!";
    const { container, getByText } = render(<App />);

    expect(container).toMatchSnapshot();
    expect(getByText(title)).toBeTruthy();

    const h1 = container.querySelector("h1");
    expect(h1.innerHTML).toContain(title);
  });
});
