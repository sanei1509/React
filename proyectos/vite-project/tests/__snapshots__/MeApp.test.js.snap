// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre componente principal REACT MeApp.js test DOM 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <h1>
        Welcome 
        Natalia Oreiro
        !!
      </h1>
      <h3>
        información sobre nuevo usuario:
      </h3>
      <div
        class="json-container"
      >
        <code>
          {"name":"Natalia","surname":"Oreiro","location":"Argrentina","born_country":"Uruguay","age":40}
        </code>
      </div>
      <h2>
        Frase Célebre
      </h2>
      <p>
        "
        No tiene frase celébre
        "
      </p>
    </div>
  </div>
</div>
`;
